conftest.py

```diff
@@ -45,6 +46,13 @@
     CACHE_IMPORTED_VM = False
 assert CACHE_IMPORTED_VM in [True, False]
 
+# Session-level tracking for failed tests and their associated hosts
+FAILED_TESTS_HOSTS = {}  # {test_nodeid: set(Host)}
+SESSION_HAS_FAILURES = False
```

Comment:
Do we really need this?  It looks like it just reflects that `FAILED_TESTS_HOSTS` has contents or not?